Dart/src/com/jetbrains/lang/dart/ide/structure/DartStructureViewModel.java

```diff
@@ -64,14 +64,15 @@ public StructureViewTreeElement getRoot() {
   @Override
   @Nullable
   public Object getCurrentEditorElement() {
+    // Note: this should return an object of type PsiElement to be compatible with the Context Info (alt+q) action.
     if (getEditor() == null) return null;
 
     final DartAnalysisServerService service = DartAnalysisServerService.getInstance(getPsiFile().getProject());
     final Outline outline = service.getOutline(getPsiFile().getVirtualFile());
     if (outline == null) return null;
 
-    final Outline result = findDeepestOutlineForOffset(getEditor().getCaretModel().getOffset(), outline);
-    return DartStructureViewElement.getValue(result);
+    final int offset = getEditor().getCaretModel().getOffset();
+    return getPsiFile().getOriginalElement().findElementAt(offset);
```

Comment:
I'm afraid this breaks `Navigate from source` functionality of the Structure View. See `DartStructureViewElement.getValue(Outline)` and its usages.
Unfortunately, we don't have tests for this. You may reproduce the regression manually: `Cmd+F12` won't focus the current element in the tree. Or toggle `Autoscroll from source` in the Structure tool window, it won't autoscroll as you move the caret in the editor.

And unfortunately, I'm not sure it's easy to have both Context Info and Structure View working.

Reply:
I believe I've fixed it.  Please take another look.